network/src/peer_registry.rs

```diff
@@ -29,14 +30,12 @@ pub struct ConnectionStatus {
     pub max_outbound: u32,
 }
 
-fn sort_then_drop<T, F>(list: &mut Vec<T>, n: usize, compare: F)
+fn sort_then_truncate<T, F>(list: &mut Vec<T>, n: usize, compare: F)
 where
     F: FnMut(&T, &T) -> std::cmp::Ordering,
 {
     list.sort_by(compare);
-    if list.len() > n {
-        list.truncate(list.len() - n);
-    }
```

Comment:
Is it a mistake?

Reply:
I think this function is intended to remove n elements from the list.

BTW, please do not mess up bug fixing with refactoring in one PR.

Reply:
OK, I dropped the associated commits.

Reply:
I see after rereading the code.